### DISCUSSION TITLE
Re-Annotation of gene coordinates

### DISCUSSION THREAD

**Discussion by ARW-UBT** [Gaius-Augustus/Augustus#371] [General]
Hello, 
I am currently analysing RNA-Seq (transcriptome) data from the ant Lasius niger to find differentially-expressed genes. The published genome assembly at NCBI does contain 18,000+ genes, which are - unfortunately - annotated only as regions from the start to the stop codon. All 5' and 3' UTRs are missing. Since our NGS library was prepared using a 3' mRNA library kit, and only one read next to the poly-A tail has been sequenced, the vast majority of mapped reads lie ouside the 'gene' feature in the not-annotated 3' UTRs. 
I have seen  that Augustus will be able to predict UTRs for the Lasisu niger genome assembly, but I am not exactly sure about the best procedure to get to this point. I would like to present my planned workflow and would appreciate any comments from the experts!

Step 1: Augustus Training: Using the published genome assembly and the corresponding CDS file from NCBI; [explanation on WebAugustus: If cDNA quality is sufficient, also a UTR training set will be created]
Question: Is it a problem, if too few genes were annotated in the published genome assembly (a first gene finding run resulted in ca. 28,000 genes found by Augustus, compared to ca. 18,000 genes at NCBI). Or what is meant by 'sufficient' quality?

Step 2: Augustus prediction: Using the published genome assembly, the training output from step 1, and RNA-Seq reads from our experiment to identify UTRs. 
Question 1: Will the output from step 1 ba a valid species parameter file? Can it be used to annotate UTRs to the genome?
Question 2: Can RNA-Seq reads (fastq files) be used additional data, and wher do have these data be provided to Augustus (I assume, fastq files are NOT 'hints data'); I used Augustus in previous projects as a tool within the OmicsBox pipeline, where RNA-Seq data can be provided to Augustus: 

General question: The species parameter list is larger than the list of species for UTR parameters. If my assumptions described above are not correct/working, is there a possibility to create Camponotus floridanus UTR parameters instead (this ant is much closer related to L. niger that Apis mellifera) and use this to re-annotate genes and UTRs on the published geneome assembly. 

I would appreciate any comments!
Best regards,

**Comment by MarioStanke** [comment]
**Q**: Is it a problem, if 18K instead of 28K genes are used for training.
**A**: No. 1-2K genes are enough. The rest could theoretically cause an issue of bias, e.g. towards highly expressed training genes, but  probably less so for you than for others.
**Question 1:** Yes, the WebAugustus training produces the config/species/ subdirectory that you can then use locally.
**Question 2:** Hints data are GFF formatted files which contain genomic coordinates, not sequences.
   For prediction with WebAugustus, you could prepare them locally: align sequences (`fastq`) to genome and produce `.bam`, then run `bam2hints`. 

We are close to releasing BRAKER3, which you could use to do everything with one command, inputting the existing annotation as protein sequences as well as the fastq files. The release should be before before Jan 15th, 2023, when we will present it on the Plant and Animal Genomes conference.

**Comment by ARW-UBT** [comment]
@MarioStanke 
Thank you for your comments and the note on the upcomming BRAKER3. 
Will BRAKER3 work in a Miniconda environment with the other tools as shown for BRAKER (https://github.com/Gaius-Augustus/BRAKER#installation)? Or should I wait for updated installation instructions?
Best regards

**Comment by KatharinaHoff** [comment]
BRAKER3 has more dependencies than BRAKER1 and BRAKER2. We will provide a docker container. Currently, the main dependency, GeneMark-ETP is not publicly available, yet. It will likely become available during or after PAG.

**Comment by ARW-UBT** [comment]
Hi,
Thank you for your comments. I assume the docker container will be available via Github. May I ask you for a short notice once it is online.
I have seen that the Braker3 presentation will be next Sunday, but unfortunately, I cannot join PAG30 only for selected presentations…

Best regards
Alfons



Von: Katharina Hoff ***@***.***>
Gesendet: Dienstag, 10. Januar 2023 15:08
An: Gaius-Augustus/Augustus ***@***.***>
Cc: Weig, Alfons ***@***.***>; Author ***@***.***>
Betreff: Re: [Gaius-Augustus/Augustus] Re-Annotation of gene coordinates (Discussion #371)


BRAKER3 has more dependencies than BRAKER1 and BRAKER2. We will provide a docker container. Currently, the main dependency, GeneMark-ETP is not publicly available, yet. It will likely become available during or after PAG.

**Reply by ARW-UBT** [reply]
Hello, 
I have found braker3 on teambraker and created a singularity build. 
According to 'print_braker3_setup.py', I looked for geneMark-ETP. At the given web page, there is GeneMart-ES/ET/EP 2.6-4 available. The archive name is gmes_linux_65. Your setup.py states that is should be something like gmetp_linux_65
Is gmes* and gmetp* the same, or do I have to wait until it will be released?

**Reply by tomasbruna** [reply]
Hi @ARW-UBT, for now, please pull GeneMark-ETP from https://github.com/gatech-genemark/GeneMark-ETP. It will eventually be added to the container; the addition is pending GeneMark-ETP PI's approval.

**Reply by ARW-UBT** [reply]
Hi @tomasbruna 
I have downloaded GeneMark-ETP and got the following error from test1.sh:
$ bash test1.sh 
Could not find /home/bt140047/GeneMark-ETP/bin/etp_release.pl. GeneMark-ETP has not been installed, correctly.
Get help with "singularity exec braker3.sif print_braker3_setup.p"

This is true, I cannot find 'etp_release.pl' in the /bin/ folder. 
"export ETP=/..."  and "export BRAKER_SIF=/..." has been set before.

Are the test*.sh scripts up-to-date?

Best regards

**Comment by KatharinaHoff** [comment]
They are not up to date. Please wait (a few days).

ARW-UBT ***@***.***> schrieb am Fr. 20. Jan. 2023 um 12:29:

**Reply by ARW-UBT** [reply]
OK!

**Reply by ARW-UBT** [reply]
Hi @KatharinaHoff , 
what is the status of GeneMark-ETP? I have seen a related discussion in the BRAKER forum and it sounded like a fixed issue ("Ok, they did change sth"). 
I just want to make sure that I did not overlook something. 
Best regards!

**Comment by KatharinaHoff** [comment]
It is unknown. It’s not in my hands… waiting for Team ETP… hang in there

ARW-UBT ***@***.***> schrieb am Di. 31. Jan. 2023 um 08:48:
